src/bundle/Resources/public/ts/components/chip.ts

```diff
@@ -0,0 +1,35 @@
+import { Base } from '../partials';
+
+interface ChipConfig {
+    onDelete?: (event: Event) => void;
+}
+
+export default class Chip extends Base {
+    private deleteButton: HTMLButtonElement | null;
+    private onDelete?: (event: Event) => void;
+
+    constructor(container: HTMLDivElement, config: ChipConfig = {}) {
+        super(container);
+
+        this.deleteButton = this._container.querySelector('.ids-chip__delete');
+        this.onDelete = config.onDelete;
+    }
+
+    private handleDelete(event: MouseEvent): void {
+        event.stopPropagation();
+        if (this.onDelete) {
+            this.onDelete(event);
+        }
+    }
+
+    private initDeleteButton(deleteButton: HTMLButtonElement): void {
+        deleteButton.addEventListener('click', this.handleDelete.bind(this));
+    }
+
+    public init(): void {
+        if (this.deleteButton) {
```

Comment:
personally I would move this if into initDeleteButton (and also use this.deleteButton directly in that method) so that this method would look only like 
```
    public init(): void {
        super.init();

        this.initDeleteButton();
    }
```
the less code here, the more you can personalize in individual methods :)
```